pkg/destroy/azure/azure.go

```diff
@@ -602,6 +606,11 @@ func deleteApplicationRegistrations(ctx context.Context, graphClient *msgraphsdk
 			errorList = append(errorList, err)
 			continue
 		}
+		// msgraphsdk can return a `nil` response even if no errors occurred
+		if resp == nil {
+			logger.Debugf("Empty response getting Application from Service Principal %s", *sp.GetDisplayName())
+			continue
+		}
 		apps := resp.GetValue()
```

Comment:
Do we have to chck for resp == nil for https://github.com/openshift/installer/blob/master/pkg/destroy/azure/azure.go#L611?

Reply:
Hard to say since the API is not very clear about it. I don't think so since in the docs they don't mention any error handling https://learn.microsoft.com/en-us/graph/sdks/create-requests?tabs=Go#delete-an-entity. Notice that `ApplicationsById` does not return an `Application` object, but an `ApplicationItemRequestBuilder` instead. I'd be very surprised if a `nil` response was returned for that.